app/services/lock-event/service.go

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2021 LimeChain Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package lock_event
+
+import (
+	"github.com/hashgraph/hedera-sdk-go/v2"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/repository"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/service"
+	lock_event "github.com/limechain/hedera-eth-bridge-validator/app/model/lock-event"
+	"github.com/limechain/hedera-eth-bridge-validator/app/model/transfer"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	SCHEDULED_MINT_TYPE     = "mint"
+	SCHEDULED_TRANSFER_TYPE = "transfer"
+	DONE                    = "DONE"
+	FAIL                    = "FAIL"
+)
+
+type Service struct {
+	bridgeAccount    hedera.AccountID
+	repository       repository.LockEvent
+	scheduledService service.Scheduled
+	logger           *log.Entry
+}
+
+func NewService(
+	bridgeAccount string,
+	repository repository.LockEvent,
+	scheduled service.Scheduled) *Service {
+
+	bridgeAcc, err := hedera.AccountIDFromString(bridgeAccount)
+	if err != nil {
+		log.Fatalf("Invalid bridge account: [%s].", bridgeAccount)
+	}
+
+	return &Service{
+		bridgeAccount:    bridgeAcc,
+		repository:       repository,
+		scheduledService: scheduled,
+		logger:           config.GetLoggerFor("Lock Event Service"),
+	}
+}
+
+func (s Service) ProcessEvent(event lock_event.LockEvent) {
+	err := s.repository.Create(event.Id, event.Amount, event.Recipient.String(), event.NativeAsset, event.WrappedAsset, event.SourceChainId.Int64(), event.TargetChainId.Int64())
```

Comment:
We can add TODO here: Based on the `sourceChain` we should trigger different logic. In one case handler for hedera scheduled transactions in another case EVM handler to publish signatures in HCS